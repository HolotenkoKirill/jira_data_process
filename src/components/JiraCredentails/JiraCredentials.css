.jira-credentails-container {
	width: 400px;
	text-align: center;
	margin: 30px auto;
  }
  
.jira-credentails-title {
	margin-top: 30px;
}
  
.jira-credentails {
	margin: 20px auto;
}
  
.jira-credentails-field {
	width: 250px !important;
}
