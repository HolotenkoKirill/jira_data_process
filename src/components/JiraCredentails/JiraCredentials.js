import React, {Fragment, useState} from 'react';
import Typography from '@material-ui/core/Typography';
import TextField from '@material-ui/core/TextField';

import './JiraCredentials.css';

const JiraCredentials = ({username, password, setUsername, setPassword}) => {
    const [isUsernameTouched, setIsUsernameTouched] = useState(false);
    const [isPasswordTouched, setIsPasswordTouched] = useState(false);

    const handleUsernameChange = (e) => {
        setUsername(e.target.value);

        if (!isUsernameTouched) {
            setIsUsernameTouched(true);
        }
    };

    const handlePasswordChange = (e) => {
        setPassword(e.target.value);

        if (!isPasswordTouched) {
            setIsPasswordTouched(true);
        }
    };

    return (
        <Fragment>
            <Typography
                variant="h6"
                gutterBottom
                align="center"
                className="jira-credentails-title"
            >
                Please, enter your JIRA credentials here!
            </Typography>
            <form
                className="jira-credentails-container"
                noValidate
                autoComplete="off"
            >
                <div className="jira-credentails">
                    <TextField
                        variant="outlined"
                        required
                        error={isUsernameTouched && !username}
                        helperText="Username can't be empty."
                        label="Usename"
                        type="text"
                        defaultValue={username}
                        className="jira-credentails-field"
                        onChange={handleUsernameChange}
                    />
                </div>
                <div className="jira-credentails">
                    <TextField
                        variant="outlined"
                        required
                        error={isPasswordTouched && !password}
                        helperText="Password can't be empty."
                        id="standard-password-input"
                        label="Password"
                        type="password"
                        defaultValue={password}
                        className="jira-credentails-field"
                        onChange={handlePasswordChange}
                    />
                </div>
            </form>
        </Fragment>
    );
};

export default JiraCredentials;
