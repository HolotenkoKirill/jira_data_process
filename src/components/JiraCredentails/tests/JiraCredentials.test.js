import React from 'react';
import {mount, shallow} from 'enzyme';
import JiraCredentials from '../JiraCredentials';
import serializer from 'enzyme-to-json/serializer';
import {act} from 'react-dom/test-utils';

expect.addSnapshotSerializer(serializer);

describe('<JiraCredentials />', () => {
    let defaultProps;

    beforeEach(() => {
        defaultProps = {
            username: 'username',
            password: 'password',
            setUsername: jest.fn(),
            setPassword: jest.fn()
        };
    });

    it('renders JiraCredentials component', () => {
        const component = shallow(<JiraCredentials {...defaultProps} />);

        expect(component).toMatchSnapshot();
    });

    it('should call setUsername function on username value change', () => {
        const component = mount(<JiraCredentials {...defaultProps} />);
        const event = {
            target: {
                value: 'other_username'
            }
        };

        act(() => {
            component
                .find('WithStyles(ForwardRef(TextField))')
                .first()
                .prop('onChange')(event);
        });

        expect(component.prop('setUsername')).toHaveBeenCalledWith(
            'other_username'
        );
    });

    it('should call setPassword function on password value change', () => {
        const component = mount(<JiraCredentials {...defaultProps} />);
        const event = {
            target: {
                value: 'other_password'
            }
        };

        act(() => {
            component
                .find('WithStyles(ForwardRef(TextField))')
                .last()
                .prop('onChange')(event);
        });

        expect(component.prop('setPassword')).toHaveBeenCalledWith(
            'other_password'
        );
    });
});
