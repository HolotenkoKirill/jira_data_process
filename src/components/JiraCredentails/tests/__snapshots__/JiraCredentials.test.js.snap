// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JiraCredentials /> renders JiraCredentials component 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Typography))
    align="center"
    className="jira-credentails-title"
    gutterBottom={true}
    variant="h6"
  >
    Please, enter your JIRA credentials here!
  </WithStyles(ForwardRef(Typography))>
  <form
    autoComplete="off"
    className="jira-credentails-container"
    noValidate={true}
  >
    <div
      className="jira-credentails"
    >
      <WithStyles(ForwardRef(TextField))
        className="jira-credentails-field"
        defaultValue="username"
        error={false}
        helperText="Username can't be empty."
        label="Usename"
        onChange={[Function]}
        required={true}
        type="text"
        variant="outlined"
      />
    </div>
    <div
      className="jira-credentails"
    >
      <WithStyles(ForwardRef(TextField))
        className="jira-credentails-field"
        defaultValue="password"
        error={false}
        helperText="Password can't be empty."
        id="standard-password-input"
        label="Password"
        onChange={[Function]}
        required={true}
        type="password"
        variant="outlined"
      />
    </div>
  </form>
</Fragment>
`;
